py/src/call.py

```diff
@@ -155,20 +159,33 @@ def loop_inner(connection, command):
     verb = command["command"]
     general_config = create_general_config(command["chain"])
 
+    at_block = command["at_block"]
     signature = command.get("signature", None)
     max_fee = command.get("max_fee", 0)
     version = command.get("version", 0)
+    gas_price = command.get("gas_price", None)
 
     timings = {}
     started_at = time.time()
-    pending_updates = command.get("pending_updates", {})
-    pending_deployed = command.get("pending_deployed", {})
+    pending_updates = command.get("pending_updates", None)
+    pending_deployed = command.get("pending_deployed", None)
 
-    # the later parts will have access to gas_price through this block_info
-    (block_info, global_root) = resolve_block(
-        connection, command["at_block"], command.get("gas_price", None)
+    fallback_to_latest = type(at_block) == bytes and (
+        pending_updates is not None or pending_deployed is not None
     )
 
+    # the later parts will have access to gas_price through this block_info
+    try:
+        (block_info, global_root) = resolve_block(connection, at_block, gas_price)
+    except NoSuchBlock:
+        if fallback_to_latest:
+            pending_updates = None
+            pending_deployed = None
+
+            (block_info, global_root) = resolve_block(connection, "latest", gas_price)
+        else:
+            raise
```

Comment:
It's a bit nasty doing this with the try-catch but I couldn't find an alternative how to do it for example with a single sql query.